hack/generator/pkg/astmodel/package_import_set.go

```diff
@@ -108,6 +113,59 @@ func (set *PackageImportSet) ApplyName(ref PackageReference, name string) {
 	}
 }
 
+// ResolveConflicts() attempts to resolve any import conflicts and returns an error if any cannot be resolved
+func (set *PackageImportSet) ResolveConflicts() error {
+
+	// Try to resolve any conflicts by renaming imports where they occur
+	for _, imports := range set.createMapByPackageName() {
+		if len(imports) <= 1 {
+			// Only one import with this name, so no conflict
+			continue
+		}
+
+		// For each import, try to assign it an alternative name
+		for _, imp := range imports {
+			alternate := set.alternatePackageNameFor(imp)
+			set.AddImport(imp.WithName(alternate))
+		}
+	}
+
+	// Check for any remaining conflicts
+	var errs []error
+	for name, imports := range set.createMapByPackageName() {
+		if len(imports) <= 1 {
+			// Only one import, so no conflict
+			continue
+		}
+
+		for _, imp := range imports {
+			err := fmt.Errorf(
```

Comment:
prefer `errors.Errorf` rather than `fmt.Errorf`?

Reply:
Changed to use `errors.Errorf()`